Gem/Code/Source/Weapons/WeaponTypes.cpp

```diff
@@ -308,17 +274,14 @@ namespace MultiplayerSample
         if (serializeContext)
         {
             serializeContext->Class<WeaponParams>()
-                ->Version(5)
+                ->Version(6)
```

Comment:
Update version needed for adding fields? I was thinking version is only useful in cases when a conversion is needed (a field was renamed and we need to intelligently move the old data into the new version)

Reply:
It's _technically_ a new version, there's definitely no harm in updating it. But keep in mind that version numbers really only affect slice data, not prefab data, so it has limited value at this point regardless.

Reply:
Really? I thought the json serializer would freak if we didn't bump versions every time we changed the serialized format of a struct? It would be awesome if we could basically forget about incrementing this..

Reply:
*sigh* no, nobody ever defined a way to handle versioning with the json serializer, because <some context about why cross-version compatibility is hard to support so we'll make it nearly impossible for anyone to implement it with our new json serialization system>.

So the json serializer doesn't use the version number. If you need to change something about your serialized data for json, you need to write a separate json converter that somehow deduces the version change using literally anything about the data _but_ the version number. :( Search the code for "JsonDeserializerContext" to see some examples.